### ISSUE TITLE
Azure AD authentication does not handle conditional access

### ISSUE THREAD

**Issue by umglurf** [kubernetes/client-go#931]
We are using conditional access to require re-authentication of MFA every 7 days on our admin users. The kubectl integration with Azure AD does not handle this, and instead gives an error, with the only way out to manually edit the kubernetes config file and remove the access token.

The error we get is
```
E0205 12:31:09.244303  774332 azure.go:154] Failed to acquire a token: unexpected error when refreshing token: refreshing token: adal: Refresh request failed. Status Code = '400'. Response body: {"error":"interaction_required","error_description":"AADSTS70043: The refresh token has expired or is invalid due to sign-in frequency checks by conditional access. The token was issued on 2021-01-21T08:51:48.2932205Z and the maximum allowed lifetime for this request is 604800.\r\nTrace ID: [redacted]\r\nCorrelation ID: [redacted]\r\nTimestamp: 2021-02-05 11:31:09Z","error_codes":[70043],"timestamp":"2021-02-05 11:31:09Z","trace_id":"[redacted]","correlation_id":"[redacted]","suberror":"token_expired"}
```

**Comment by brendandburns** [comment]
What version of kubectl are you running? I believe that this was fixed in Kubernetes 1.19

https://github.com/kubernetes/kubernetes/pull/86481

You can run `kubectl version` to determine your version. If this is persisting in 1.19 or 1.20 versions of kubectl, please let us know.

**Comment by umglurf** [comment]
Hi, this was with version 1.20.2

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"archive", BuildDate:"2021-01-14T10:55:09Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
```
I'll let you know when it happens again.

**Comment by stigok** [comment]
I hit the same issue today
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"archive", BuildDate:"2021-01-13T22:07:54Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by umglurf** [comment]
And now it happened again to med, still on 1.20.2

**Comment by tdihp** [comment]
Reproducing the symptom.

**Comment by umglurf** [comment]
I have tested using [Kubelogin](https://github.com/Azure/Kubelogin) and so far it has worked without problems.

**Comment by ronaknnathani** [comment]
Is there a plan to fix this in client-go so that using `kubectl` on expired tokens doesn't throw an error?

**Comment by ronaknnathani** [comment]
Nvm. I looked at the linked PR (https://github.com/kubernetes/kubernetes/pull/102063) it seems to have been solved in v1.22+.
